src/main/java/com/booleanuk/Poker.java

```diff
@@ -28,4 +79,120 @@ public String extensionThreeMethods() {
+
+    private int getValue(String card) {
+        return this.valuesOfCards.get(card);
+    }
+
+    private int getValueOfBestCard(String[] hand) {
+        int tempValue;
+        int cardValue = 0;
+        for (String card : hand) {
+            tempValue = getValue(card);
+            if (tempValue > cardValue) {
+                cardValue = tempValue;
+            }
+        }
+        return cardValue;
+    }
+
+    private String[] bestHand(String[] firstHand, String[] secondHand) {
+
+        if (isThreeOfAKind(firstHand) && isThreeOfAKind(secondHand)) {
+            return bestHighestCard(firstHand, secondHand);
+        } else if (isThreeOfAKind(firstHand) && !isThreeOfAKind(secondHand)) {
+            return firstHand;
+        } else if (!isThreeOfAKind(firstHand) && isThreeOfAKind(secondHand)) {
+            return secondHand;
+        }
+
+        if (containsAPair(firstHand) && containsAPair(secondHand)) {
+            return bestPair(firstHand, secondHand);
+        } else if (containsAPair(firstHand) && !containsAPair(secondHand)) {
+            return firstHand;
+        } else if (!containsAPair(firstHand) && containsAPair(secondHand)) {
+            return secondHand;
+        }
+        return new String[]{};
+    }
+
+    private String[] bestHighestCard(String[] firstHand, String[] secondHand) {
+        int first = getValueOfBestCard(firstHand);
+        int second = getValueOfBestCard(secondHand);
+
+        if (first > second) {
+            return firstHand;
+        }
+        return secondHand;
+    }
+
+    private String[] bestPair(String[] firstHand, String[] secondHand) {
+        int first = pairValue(firstHand);
+        int second = pairValue(secondHand);
+
+        if (first > second) {
+            return firstHand;
+        }
+        return secondHand;
+    }
+
+    private int pairValue(String[] hand) {
+        if (hand.length == 2) {
+            return this.valuesOfCards.get(hand[firstCard]);
+        }
+        if (hand[firstCard].equals(hand[secondCard])) {
+            return this.valuesOfCards.get(hand[firstCard]);
+        }
+        else if (hand[firstCard].equals(hand[thirdCard])) {
+            return this.valuesOfCards.get(hand[firstCard]);
+        }
+        else if (hand[thirdCard].equals(hand[secondCard])) {
+            return this.valuesOfCards.get(hand[thirdCard]);
+        }
+        return -1;
+    }
+
+    private void setValuesOfCards() {
+        this.valuesOfCards.put("2", 2);
+        this.valuesOfCards.put("3", 3);
+        this.valuesOfCards.put("4", 4);
+        this.valuesOfCards.put("5", 5);
+        this.valuesOfCards.put("6", 6);
+        this.valuesOfCards.put("7", 7);
+        this.valuesOfCards.put("8", 8);
+        this.valuesOfCards.put("9", 9);
+        this.valuesOfCards.put("10", 10);
+        this.valuesOfCards.put("J", 11);
+        this.valuesOfCards.put("Q", 12);
+        this.valuesOfCards.put("K", 13);
+        this.valuesOfCards.put("A", 14);
+    }
 }
```

Comment:
Your implementation effectively addresses the core task and first extension, and your approach to handling three-card hands in Extension 2 shows thoughtful problem-solving. Your method structure and use of Java concepts are clear and well-organized. Nice work.